### ISSUE TITLE
Exception Raised in Utils.vala

### ISSUE THREAD

**Issue by MessedUpHare** [tkashkin/GameHub#30]
An Exception occurs against a build of the latest commit in Utils.vala
I don't know vala, but assume it's trying to split a string object called url on "?", then split the first returned item on ".". I'm not sure what'd cause this exception - assuming a string without either of these characters would just return the string as is... Perhaps it's null or not instantiated for some reason?

[gdb.txt](https://github.com/tkashkin/GameHub/files/2284702/gdb.txt)

Application output:
```
Application exited
Registering Stacktrace 0.9.4
[INFO 21:46:48.448545] Application.vala:154: GameHub version: 0.6.2
[INFO 21:46:48.448579] Application.vala:156: Kernel version: 4.17.14-arch1-1-ARCH
(com.github.tkashkin.gamehub:5608): Gtk-DEBUG: 21:46:48.575: Connecting to session manager
** (WebKitWebProcess:5660): DEBUG: 21:46:49.426: Name com.canonical.AppMenu.Registrar does not exist on the session bus

[FATAL 21:48:06.375578] [GLib] g_strsplit: assertion 'string != NULL' failed
An error occured (SIGSEGV)
   The reason is likely a null reference being used.
   Note: no file path and line numbers can be retrieved. Are you sure you added -g -X -rdynamic to valac command line?

    #1  <unknown>   in 'g_main_context_dispatch'
        at /usr/lib/libglib-2.0.so.0
    #2  <unknown>   in 'g_main_context_iteration'
        at /usr/lib/libglib-2.0.so.0
    #3  <unknown>   in 'g_application_run'
        at /usr/lib/libgio-2.0.so.0
    #4  <unknown>   in '__libc_start_main'
        at 
```

**Comment by tkashkin** [comment]
Yes, it's trying to get image extension from its url. I'm not sure why exactly url is null here, I'll add some null checks.

Btw latest version should be 0.7.0, but your log is from 0.6.2.

**Comment by tkashkin** [comment]
Should be fixed in eff0aab.

**Comment by MessedUpHare** [comment]
@tkashkin  - Now I don't get this fault.
Thanks very much!
